### PR TITLE
chore: initial github codespace configuration

### LINKED ISSUE #75: Add configuration for GitHub codespace
[GitHub codespaces](https://github.com/features/codespaces) make it easy to onboard new contributors. We should add the configuration necessary for a GitHub codespace, including all the tools necessary for development (helm, chart-testing, helm-docs, etc.)

### PR DESCRIPTION
**Pull request by JamieMagee**
Builds on much of the work done by @viceice for creating dependency toolchain docker images and extends this to install development toolchains as well. This is a draft PR, because I want to get feedback on the approach, before going any further.

Closes #75

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+COPY tools /usr/local/build/
+
+# renovate: datasource=github-releases depName=kubernetes/kubectl
+ARG KUBECTL_VERSION=1.19.3
+RUN install-tool kubectl
+
+# FROM mcr.microsoft.com/vscode/devcontainers/base:0.148.0-bionic
+# ARG HELM_VERSION=3.4.0
+
+# COPY --from=helm /usr/local/helm/${HELM_VERSION}/helm /usr/local/bin/helm
```
Comment (viceice) [inline .devcontainer/Dockerfile:12]:
```suggestion
# renovate: datasource=github-releases depName=helm/helm
ARG HELM_VERSION=3.4.0
RUN install-tool helm
```

needs renovate config change:
```json
"packageRules": [
    {
      "packageNames": ["helm/helm"],
      "extractVersion": "^v(?<version>.*)$"
    }
  ]
```

File: .devcontainer/tools/kubectl.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+
+check_semver $KUBECTL_VERSION
+
+curl -sSL -o /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v"$KUBECTL_VERSION"/bin/linux/amd64/kubectl
```
Comment (viceice) [inline .devcontainer/tools/kubectl.sh:7]:
```suggestion
curl -sSL -o /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl
```

File: .devcontainer/tools/kubectl.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+chmod +x /usr/local/bin/kubectl
```
Comment (viceice) [inline .devcontainer/tools/kubectl.sh:8]:
```suggestion
chmod +x /usr/local/bin/kubectl
kubectl version --client=true
```

**Review by viceice — COMMENTED** [review]
I can't really test it, as i've no access to code spaces 🙃

**Comment by JamieMagee** [comment]
> I can't really test it, as i've no access to code spaces

You can test locally using [vscode containers](https://code.visualstudio.com/docs/remote/containers). It's exactly the same technology, just for GitHub codespaces vscode is running in your browser.

**Comment by viceice** [comment]
So I think we need the helm test tools to lint /test locally

**Comment by JamieMagee** [comment]
> So I think we need the helm test tools to lint /test locally

I agree, but I didn't want to put the effort in to package them before getting your opinion on this approach. What do you think about reusing your docker buildpack work for dev containers here, and potentially in other repositories in the renovatebot organisation?

**Comment by viceice** [comment]
I like that idea. Maybe we should move the build to our ci and publish to ghcr. So the user doesn't need to / wait for rebuild the container every time. 🤔

File: .devcontainer/tools/kubectl.sh (hunk 2 above)
Comment (JamieMagee) [reply .devcontainer/tools/kubectl.sh:7]: Is this correct? Shellcheck doesn't seem to agree: https://github.com/koalaman/shellcheck/wiki/SC2086

Comment (viceice) [reply .devcontainer/tools/kubectl.sh:7]: Ok, i see, so quoted the full url. 🙃

File: .devcontainer/tools/chart_testing.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (viceice) [inline .devcontainer/tools/chart_testing.sh:1]: Why not using `chart-testing` as name? Or better `helm-chart-testing`? 🤔

File: .devcontainer/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ARG HELM_VERSION=3.4.0
+RUN install-tool helm
+
+# renovate: datasource=github-releases depName=helm/chart-testing
+ARG CHART_TESTING_VERSION=3.3.0
+RUN install-tool chart_testing
+
+# renovate: datasource=github-releases depName=norwoodj/helm-docs
+ARG HELM_DOCS_VERSION=1.4.0
+RUN install-tool helm_docs
```
Comment (viceice) [inline .devcontainer/Dockerfile:19]: Add newline 🙃

**Review by viceice — CHANGES_REQUESTED** [review]

File: .devcontainer/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,19 @@
+FROM renovate/buildpack:2
```
Comment (viceice) [inline .devcontainer/Dockerfile:1]:
```suggestion
FROM renovate/buildpack:2-focal
```
Use ubuntu focal to have newer python.

File: .devcontainer/tools/chart_testing.sh (hunk 4 above)
Comment (JamieMagee) [reply .devcontainer/tools/chart_testing.sh:1]: I hit some issues with trying to import this script while using `-`. `_` worked fine though.

**Comment by viceice** [comment]
ref renovatebot/docker-buildpack#88

File: .devcontainer/tools/chart_testing.sh (hunk 4 above)
Comment (viceice) [reply .devcontainer/tools/chart_testing.sh:1]: Was hit by the same issue [here](https://github.com/visualon/docker-images/blob/3ce332d1580911efb3b7128ea8744f962ecc7cbc/docker/github-runner/Dockerfile#L31) 😛 

So simply fixed it in renovatebot/docker-buildpack#88 🙃

File: .devcontainer/tools/kubectl.sh (hunk 2 above)
Comment (viceice) [reply .devcontainer/tools/kubectl.sh:7]: I'm nor sure if we even need the escape, as we check the value right before 🤔

**Comment by JamieMagee** [comment]
I've got this (mostly working) now. The 2 missing pieces are:
1. `helm` is not added to `PATH` correctly (but it is installed correctly)
2. Connection to the host `docker.sock` is not initialised correctly

Even with the issues above, I still think this is useful, and they can be addressed later.

File: .devcontainer/devcontainer.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "renovatebot/helm-charts",
+  "dockerFile": "Dockerfile",
+  "remoteUser": "ubuntu",
+  "runArgs": ["--init"],
+  "mounts": ["source=/var/run/docker.sock,target=/var/run/docker-host.sock,type=bind"],
```
Comment (viceice) [inline .devcontainer/devcontainer.json:6]:
```suggestion
  "mounts": ["source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind"],
```

Why changed path?

File: renovate.json [hunk 8]
```diff
@@ -44,6 +44,14 @@
...
       "automerge": true
+    },
+    {
+      "packageNames": ["helm/helm"],
+      "extractVersion": "^v(?<version>.*)$"
+    },
+    {
+      "packageNames": ["kubernetes/kubectl"],
+      "extractVersion": "^v(?<version>.*)$"
     }
```
Comment (viceice) [inline renovate.json:55]:
```suggestion
    {
      "packageNames": ["helm/helm", "kubernetes/kubectl"],
      "extractVersion": "^v(?<version>.*)$"
    }
```

Simplify 🙃

**Review by viceice — COMMENTED** [review]
Strange, helm should be installed as `/usr/local/bin/helm` maybe something is overriding the path with wrong values

File: .devcontainer/devcontainer.json (hunk 7 above)
Comment (JamieMagee) [reply .devcontainer/devcontainer.json:6]: It was part of the guide [here](https://github.com/microsoft/vscode-dev-containers/blob/master/script-library/docs/docker.md). I couldn't get it to work, due to permissions, so it's a moot point.

**Comment by JamieMagee** [comment]
Helm is installed to `/usr/local/helm/3.4.0`

**Pull request merged**